From 206cba36b7b185aa441c9c2f5c9e8469a6e09911 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Tue, 6 Apr 2021 15:44:12 -0500
Subject: [PATCH 12/13] HSD #22012814951-1: intel_fcs: don't report error with
 rejected service

commit  6375c74c6db993c346d51120896122f96f23a159 from
https://github.com/altera-opensource/linux-socfpga.git

Change to not report mailbox error when service request was rejected.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/intel_fcs.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index f5fb1da2110e..f203113c13a5 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -55,7 +55,6 @@ static void fcs_data_callback(struct stratix10_svc_client *client,
 			      struct stratix10_svc_cb_data *data)
 {
 	struct intel_fcs_priv *priv = client->priv;
-	unsigned int *status = (unsigned int *)data->kaddr1;
 
 	if ((data->status == BIT(SVC_STATUS_OK)) ||
 	    (data->status == BIT(SVC_STATUS_COMPLETED))) {
@@ -69,8 +68,8 @@ static void fcs_data_callback(struct stratix10_svc_client *client,
 		priv->size = (data->kaddr3) ?
 			*((unsigned int *)data->kaddr3) : 0;
 	} else {
-		dev_err(client->dev, "failed, mbox_error=0x%x\n", *status);
-		priv->status = *status;
+		dev_warn(client->dev, "rejected, invalid param\n");
+		priv->status = -EINVAL;
 		priv->kbuf = NULL;
 		priv->size = 0;
 	}
-- 
2.17.1

